packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -108,71 +109,72 @@ export class ZkEvmProvider implements Provider {
 
   private isLoggedIn(): this is LoggedInZkEvmProvider {
     return this.magicProvider !== undefined
-      && this.user !== undefined
+      && this.zkevmAddress !== undefined
       && this.relayerClient !== undefined
       && this.guardianClient !== undefined;
   }
 
   private async performRequest(request: RequestArguments): Promise<any> {
     switch (request.method) {
       case 'eth_requestAccounts': {
+        const existingUser = await this.authManager.getUser() as UserZkEvm;
         if (this.isLoggedIn()) {
-          return [this.user.zkEvm.ethAddress];
+          return [existingUser.zkEvm.ethAddress];
         }
```

Comment:
Can we move the call to `getUser` back into `loginZkEvmUser` please? We can also update this return to be `return [this.zkevmAddress];`